source/common/formatter/substitution_formatter.cc

```diff
@@ -760,6 +760,14 @@ class StreamInfoAddressFieldExtractor : public StreamInfoFormatter::FieldExtract
       return unspecifiedValue();
     }
 
+    if (extraction_type_ == StreamInfoFormatter::StreamInfoAddressFieldExtractionType::JustPort) {
+      auto port = StreamInfo::Utility::extractDownstreamAddressJustPort(*address);
```

Comment:
I think we probably need to add a runtime flag for this to give operators time to handle the change. Otherwise it's going to break someone's log processing.

Reply:
+1, there could be some log process assuming it is string, but now it will be numbers.
You can add runtime guard [here](https://github.com/envoyproxy/envoy/blob/b41adf18d0419d4c34cc153bf53b3d27295068a3/source/common/runtime/runtime_features.cc):

Reply:
Thank you for the suggestion, I have added a runtime guard.